publication-commons/src/main/java/no/unit/nva/publication/model/storage/ResourceDao.java

```diff
@@ -153,6 +161,14 @@ public void setDoi(URI doi) {
         this.doi = doi;
     }
 
+    public List<ImportDetail> getImportDetails() {
+        return nonNull(importDetails) ? importDetails : Collections.emptyList();
+    }
+
+    public void setImportDetails(Collection<ImportDetail> importDetails) {
+        this.importDetails = new ArrayList<>(importDetails);
```

Comment:
This one (and the other `setImportDetails`-setters in this repo) doesn't have the restriction that Publication has (from the datamodel). Here, one can remove existing `importDetails`. Not sure if this will be a problem.